src/modules/tracker.js

```diff
@@ -63,6 +64,36 @@ function applyParams(parameters, options) {
     if (pathname) {
       aggregateParams.origin_referrer += pathname;
     }
+
+    if (search) {
+      try {
+        const searchParams = new URLSearchParams(search);
+        const utmSource = searchParams.get('utm_source');
+        const utmMedium = searchParams.get('utm_medium');
+        const utmCampaign = searchParams.get('utm_campaign');
+
+        // Add UTM parameters to origin_referrer if they exist
+        if (utmSource || utmMedium || utmCampaign) {
+          aggregateParams.origin_referrer += '?';
+
+          if (utmSource) {
+            aggregateParams.origin_referrer += `utm_source=${utmSource}`;
+          }
+
+          if (utmMedium) {
+            if (utmSource) aggregateParams.origin_referrer += '&';
+            aggregateParams.origin_referrer += `utm_medium=${utmMedium}`;
+          }
+
+          if (utmCampaign) {
+            if (utmSource || utmMedium) aggregateParams.origin_referrer += '&';
+            aggregateParams.origin_referrer += `utm_campaign=${utmCampaign}`;
+          }
```

Comment:
Might I suggest an alternative?

```js
    const utmParamKeys = ['utm_source', 'utm_medium', 'utm_campaign'];
    const utmQueryParamStrArr = [];

    utmParamKeys.forEach((key) => {
      const utmParamValue = searchParams.get(key);

      if (utmParamValue) {
        utmQueryParamStrArr.push(`${key}=${utmParamValue}`);
      }
    });

    if (utmQueryParamStrArr.length) {
      aggregateParams.origin_referrer += `?${utmQueryParamStrArr.join('&')}`
    }
```

Not to nitpick on implementation, but something like this would make it easier if we needed to add in more utm fields in the future

Reply:
I like that